deepspeed/runtime/utils.py

```diff
@@ -125,9 +130,17 @@ def has_overflow(self, params):
                                          op=torch.distributed.ReduceOp.MAX,
                                          group=torch.distributed.group.WORLD)
         elif self.mpu is not None:
+            if self.deepspeed.pipeline_enable_backward_allreduce is False:
```

Comment:
This will crash if `self.deepspeed = None` right?

Reply:
Yes you are right. I added checks about whether self.deepspeed exist, and whether it is PipelineEngine or not.